File: 01.js
const fs = require('fs')
const readline = require('readline')
const events = require('events')
input_file = "01input.txt"

async function processLineByLine(f, callback){
    try {
      const rl = readline.createInterface({
        input: fs.createReadStream(f),
        crlfDelay: Infinity
      })
      rl.on('line', callback)
      await events.once(rl, 'close')
    }
    catch (err) {
      console.error(err)
    }
}

function sum(arr){
    return arr.reduce((accumulator, value) => {
            return accumulator + value
            }, 0)
}

async function main(ex, file){
    let max_sum, elf_sum, elf_sums
    max_sum = 0
    elf_sum = 0
    elf_sums = []
    await processLineByLine(file, line => {
        if (line != ""){
            elf_sum += parseInt(line)
        }
        else {
            elf_sums.push(elf_sum)
            elf_sum = 0
        }
    })
    if(ex === 1){
        console.log(Math.max(...elf_sums))
    }
    else{
        elf_sums.sort((a, b) => {
            return a - b
        })
        console.log(sum(elf_sums.slice(-3)))
    }
}

async function doit(){
    await main(1, input_file)
    await main(2, input_file)
}

doit()

File: 01.py
import os
import sys

curr_dir = os.path.dirname(os.path.realpath(__file__))
input_file = f"{curr_dir}/01input.txt"

def main(ex, file):
    with open(file, "r") as f:
        max_sum = 0
        elf_sum = 0
        elf_sums = []
        for line in f:
            line = line.strip()
            if line != "":
                elf_sum += int(line)
            else:
                elf_sums.append(elf_sum)
                elf_sum = 0
        elf_sums.sort()
        if ex == 1:
            print(elf_sums[-1])
        else:
            print(sum(elf_sums[-3:]))

main(1, input_file)
main(2, input_file)
